### ISSUE TITLE
"\\" seems to trigger an infinite loop in strv_split_full

### ISSUE THREAD

**Issue by evverx** [systemd/systemd#15305]
Thanks to a new fuzz target that was added in https://github.com/systemd/systemd/pull/15226 it was discovered that "\\\\" can trigger an infinite loop in `strv_split_full` (which seems to have never been fuzzed before according to https://storage.googleapis.com/oss-fuzz-coverage/systemd/reports/20200401/linux/src/systemd/src/basic/strv.c.html#L247). It can be reproduced by applying the following patch:
```diff
diff --git a/src/test/test-strv.c b/src/test/test-strv.c
index 68c128cf80..70030ad0a1 100644
--- a/src/test/test-strv.c
+++ b/src/test/test-strv.c
@@ -307,6 +307,8 @@ static void test_strv_split(void) {
         l = strv_split_full("    'one'  \"  two\t three \"' four  five", NULL, SPLIT_QUOTES | SPLIT_RELAX);
         assert_se(l);
         assert_se(strv_equal(l, (char**) input_table_quoted));
+
+        l = strv_split_full("\\", NULL, SPLIT_QUOTES | SPLIT_RELAX);
 }
```

**Comment by poettering** [comment]
ouch

**Comment by poettering** [comment]
Fix waiting in #15309
